packages/drivers/odsp-driver/src/epochTracker.ts

```diff
@@ -298,27 +300,48 @@ export class EpochTracker implements IPersistedFileCache {
 			.catch((error) => {
 				// If the error is about location redirection, then we need to generate new resolved url with correct
 				// location info.
-				if (
-					isFluidError(error) &&
-					error.errorType === OdspErrorTypes.fileNotFoundOrAccessDeniedError
-				) {
-					const redirectLocation = (error as IOdspErrorAugmentations).redirectLocation;
-					if (redirectLocation !== undefined) {
-						const redirectUrl: IOdspResolvedUrl = patchOdspResolvedUrl(
-							this.fileEntry.resolvedUrl,
-							redirectLocation,
-						);
-						const locationRedirectionError = new LocationRedirectionError(
+				if (isFluidError(error)) {
+					if (error.errorType === OdspErrorTypes.fileNotFoundOrAccessDeniedError) {
+						const redirectLocation = (error as IOdspErrorAugmentations)
+							.redirectLocation;
+						if (redirectLocation !== undefined) {
+							const redirectUrl: IOdspResolvedUrl = patchOdspResolvedUrl(
+								this.fileEntry.resolvedUrl,
+								redirectLocation,
+							);
+							const locationRedirectionError = new LocationRedirectionError(
+								error.message,
+								redirectUrl,
+								{ driverVersion, redirectLocation },
+							);
+							locationRedirectionError.addTelemetryProperties(
+								error.getTelemetryProperties(),
+							);
+							throw locationRedirectionError;
+						}
+					} else if (
+						error.errorType === OdspErrorTypes.throttlingError &&
+						this.hostPolicy?.disableRetriesOnStorageThrottlingError
+					) {
+						const nonRetriableThrottlingError = new NonRetryableError(
 							error.message,
-							redirectUrl,
-							{ driverVersion, redirectLocation },
+							OdspErrorTypes.throttlingError,
+							// retryAfterMs is still provided so that client app can can still use it.
+							{
+								retryAfterMs:
```

Comment:
Why not use `retryAfterSeconds` itself as the property name instead to introducing a new one?